snapper/worker.py

```diff
@@ -79,70 +81,30 @@ def host_worker(host_queue, file_queue, timeout,
             else:
                 logging.debug("%s is unreachable or timed out", host)
 
-
-def capture_snaps(urls, outpath, timeout, num_workers,
-                  user_agent, result, phantomjs_binary, task):
-    css_output_path = Path(outpath) / "css"
-    js_output_path = Path(outpath) / "js"
-    images_output_path = Path(outpath) / "images"
-
-    os.makedirs(css_output_path)
-    os.makedirs(js_output_path)
-    os.makedirs(images_output_path)
-
-    css_template_path = Path(__file__).parent / "templates" / "css"
-    js_template_path = Path(__file__).parent / "templates" / "js"
-    shutil.copyfile(
-        css_template_path / "materialize.min.css",
-        css_output_path / "materialize.min.css"
-    )
-    shutil.copyfile(
-        js_template_path / "jquery.min.js",
-        js_output_path / "jquery.min.js"
-    )
-    shutil.copyfile(
-        js_template_path / "materialize.min.js",
-        js_output_path / "materialize.min.js"
-    )
-
-    host_queue = Queue()
-    file_queue = Queue()
-
-    workers = []
-    for host in urls:
-        host_queue.put(host)
-    for i in range(num_workers):
-        p = Process(target=host_worker, args=(host_queue, file_queue, timeout,
-                    user_agent, outpath, phantomjs_binary))
-        workers.append(p)
-        p.start()
-
-    try:
-        for worker in workers:
-          worker.join()
-    except KeyboardInterrupt:
-        for worker in workers:
-          worker.terminate()
-          worker.join()
-        sys.exit()
-
-    sets_of_six = []
-    count = 0
-    urls = {}
-    while not file_queue.empty():
-        if count == 6:
+    task.left = task.left-1
```

Comment:
This needs to be done under mutex, in Python it's threading.Lock() or multiprocessing.Lock(). Otherwise it may be a race condition.

Reply:
will look into it